sys/vfs_syscalls.c

```diff
@@ -173,3 +186,27 @@ int sys_fstat(thread_t *td, syscall_args_t *args) {
     return error;
   return 0;
 }
+
+int sys_mount(thread_t *td, syscall_args_t *args) {
+  char *user_fsysname = (char *)args->args[0];
+  char *user_pathname = (char *)args->args[1];
+
+  int error = 0;
+  char fsysname[256];
```

Comment:
We must stop using kernel stack so carelessly... after all we have only a page (or two?) of it! Please look for reference at FreeBSD [sys_mount](http://bxr.su/FreeBSD/sys/kern/vfs_mount.c#731) – the actually use `M_TEMP` memory pool to `copyinstr`.